### PR TITLE
[ggphp] feat(metadata): Add gapic_metadata.json generation, CLI flag

### PR DESCRIPTION
**Pull request by miraleung**

### CODE REVIEW

File: src/Generation/GapicMetadataGenerator.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                $libraryClient->setLibraryClient($serviceDetails->gapicClientType->name);
+                $libraryClient->setRpcs(
+                    $serviceDetails->methods->toArray(
+                        fn ($method) => $method->name,
+                        fn ($method) => (new GapicMetadata\MethodList())->setMethods([$method->methodName])
+                    )
+                );
+                $transport = new GapicMetadata\ServiceForTransport();
+                $transport->setClients(['grpc' => $libraryClient]);
+                $gapicMetadataServices[$serviceDetails->serviceName] = $transport;
```
Comment (vchudnov-g) [inline src/Generation/GapicMetadataGenerator.php:83]: This should NOT be the fully qualified service name, but rather the part after the value in `protoPackage` (see my other comment)

File: tests/Integration/goldens/asset/src/V1/gapic_metadata.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
+{
+    "schema": "1.0",
+    "comment": "This file maps proto services\/RPCs to the corresponding library clients\/methods",
+    "language": "php",
+    "protoPackage": "google.cloud.asset.v1",
+    "libraryPackage": "Google\\Cloud\\Asset\\V1",
+    "services": {
+        "google.cloud.asset.v1.AssetService": {
```
Comment (vchudnov-g) [inline tests/Integration/goldens/asset/src/V1/gapic_metadata.json:8]: This should NOT be the fully qualified service name, but rather the part after the value in `protoPackage`, ie: simply `"AssetService"` in this case

**Review by vchudnov-g — CHANGES_REQUESTED** [review]
Thanks for doing this! One fix requested on the service name.

File: src/Generation/GapicMetadataGenerator.php (hunk 1 above)
Comment (miraleung) [reply src/Generation/GapicMetadataGenerator.php]: Done.

File: tests/Integration/goldens/asset/src/V1/gapic_metadata.json (hunk 2 above)
Comment (miraleung) [reply tests/Integration/goldens/asset/src/V1/gapic_metadata.json]: Done.

**Comment by miraleung** [comment]
Merging given the fixes and OOO.

**Pull request merged**